src/datachain/catalog/catalog.py

```diff
@@ -1438,26 +1439,34 @@ def get_file_signals(
         # to open object
         return next(iter(file_signals_values.values()))
 
-    def open_object(
-        self,
-        dataset_name: str,
-        dataset_version: int,
-        row: RowDict,
-        use_cache: bool = True,
-        **config: Any,
-    ):
+    def get_file_from_row(
+        self, dataset_name: str, dataset_version: int, row: RowDict, signal_name: str
+    ) -> "File":
+        """
+        Function that returns specific file signal from dataset row by name.
+        """
         from datachain.lib.file import File
+        from datachain.lib.signal_schema import DEFAULT_DELIMITER, SignalSchema
+
+        version = self.get_dataset(dataset_name).get_version(dataset_version)
+        schema = SignalSchema.deserialize(version.feature_schema)
+
+        if signal_name not in schema.get_signals(File):
+            raise RuntimeError(
+                f"File signal with path {signal_name} not found in ",
+                f"dataset {dataset_name}@v{dataset_version} signals schema",
+            )
 
-        file_signals = self.get_file_signals(dataset_name, dataset_version, row)
-        if not file_signals:
-            raise RuntimeError("Cannot open object without file signals")
+        prefix = signal_name.replace(".", DEFAULT_DELIMITER) + DEFAULT_DELIMITER
```

Comment:
Should we move this to some helper method? I've seen same line in another parts of the code.

Reply:
I've just remove other place where it was used as that whole method is not used any more so it's removed. I will see if it makes sense to put it to utils anyway though